tests/parsers/winreg.py

```diff
@@ -102,6 +105,78 @@ def testParseSystem(self):
     parser_chain = self._PluginNameToParserChain('windows_services')
     self.assertEqual(parser_chains.get(parser_chain, 0), 831)
 
+  @shared_test_lib.skipUnlessHasTestFile(['SYSTEM'])
+  def testParseSystemWithArtifactsFilter(self):
+    """Tests the Parse function on a SYSTEM file with artifact filters."""
+    parser = winreg.WinRegistryParser()
+    knowledge_base = knowledge_base_engine.KnowledgeBase()
+    with tempfile.NamedTemporaryFile(delete=False) as temp_file:
+      test_filter_file = artifacts_filter_file.ArtifactsFilterFile(
+          temp_file.name, knowledge_base)
+      temp_file.write(b'name: TestRegistryKey\n')
```

Comment:
move the data to write to a string or list and write this in a single call, also looks like you can move the creation of the test data out of the with

Reply:
Done